firedrake/ensemble.py

```diff
@@ -35,32 +35,123 @@ def __init__(self, comm, M):
         assert self.comm.size == M
         assert self.ensemble_comm.size == (size // M)
 
+    def __del__(self):
+        if hasattr(self, "comm"):
+            self.comm.Free()
+            del self.comm
+        if hasattr(self, "ensemble_comm"):
+            self.ensemble_comm.Free()
+            del self.ensemble_comm
+
+    def _check_function(self, f, g=None):
+        """
+        Check if function f (and possibly a second function g) is a valid argument for ensemble mpi routines
+
+        :arg f: The function to check
+        :arg g: Second function to check
+        :raises ValueError: if function communicators mismatch each other or the ensemble spatial communicator, or is the functions are in different spaces
```

Comment:
Wrap long lines (don't forget to indent subsequent ones (by at least 4 spaces) so it's treated as part of the same block).

Reply:
Updated now